medusa/providers/torrent/json/eztv.py

```diff
@@ -0,0 +1,150 @@
+# coding=utf-8
+
+"""Provider code for EZTV."""
+
+from __future__ import unicode_literals
+
+import logging
+import time
+
+from medusa import tv
+from medusa.helper.common import convert_size
+from medusa.indexers.utils import mappings
+from medusa.logger.adapters.style import BraceAdapter
+from medusa.providers.torrent.torrent_provider import TorrentProvider
+
+log = BraceAdapter(logging.getLogger(__name__))
+log.logger.addHandler(logging.NullHandler())
+
+
+class EztvProvider(TorrentProvider):
+    """EZTV Torrent provider."""
+
+    def __init__(self):
+        """Initialize the class."""
+        super(EztvProvider, self).__init__('Eztv')
+
+        # Credentials
+        self.public = True
+
+        # URLs
+        self.url = 'https://eztv.io'
+        self.urls = {
+            'api': 'https://eztv.io/api/get-torrents',
+        }
+
+        # Proper Strings
+
+        # Miscellaneous Options
+
+        # Cache
+        self.cache = tv.Cache(self, min_time=15)
+
+    def search(self, search_strings, age=0, ep_obj=None, **kwargs):
+        """
+        Search a provider and parse the results.
+
+        :param search_strings: A dict with mode (key) and the search value (value)
+        :param age: Not used
+        :param ep_obj: Not used
+        :returns: A list of search results (structure)
+        """
+        results = []
+
+        # Search Params
+        search_params = {
+            'imdb_id': '*',
+            # 'limit': 30,
+        }
+
+        for mode in search_strings:
+            log.debug('Search mode: {0}', mode)
+
+            for search_string in search_strings[mode]:
+
+                if mode != 'RSS':
+                    imdb_id = self.series.externals.get(mappings[10])
+                    imdb_id = imdb_id[2:]  # strip two tt's of id as they are not used by eztv
+                    if imdb_id:
+                        search_params['imdb_id'] = imdb_id
+                        log.debug('Search string (IMDb ID): {imdb_id}',
+                                  {'imdb_id': imdb_id})
+                    else:
+                        log.debug('IMDb ID not found')
+                        continue
+
+                # Maximum requests allowed are 1req/2sec
+                # Changing to 5 because of server clock desync
+                time.sleep(5)

```

Comment:
Is this really necessary?

Reply:
Actually yes. I am working back from 30sec. It is like a debounce routine which is needed else eztv will sometimes not be able to search due to the limitation of search request they added to their API.

Reply:
Hmmz seems to be working without it. This was from their old API. I removed it but it requires testing. I will look into that and post my findings. Either way in or out will not harm something.

Reply:
Actually i just tested without the time.sleep and manual seacrh breaks. So it is still needed for manual search

Reply:
Just tested and it works without the sleep()